src/main/webapp/js/components/settings/csv.jsx

```diff
@@ -0,0 +1,319 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { FieldArray } from 'react-final-form-arrays';
+import { Form } from 'react-final-form';
+import arrayMutators from 'final-form-arrays';
+
+import 'react-table/react-table.css';
+
+import Select from '../../utils/inputs/select';
+import {
+  createCsvConfig,
+  deleteCsvConfig,
+  fetchCsvConfig,
+  saveCsvConfig,
+} from '../../actions';
+import renderFormField from '../../utils/form/form-utils';
+
+const ENTITY_TYPE = [
+  { label: 'Vaccinee', value: 'vaccinee' },
+  { label: 'Visit', value: 'visit' },
+  { label: 'Site', value: 'site' },
+];
+
+class CsvConfig extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      csv: this.props.csvConfig.csv,
+      isSelectDisabled: false,
+    };
+  }
+
+
+  onSubmit(csvFields, entity) {
+    this.setState({
+      csv: {
+        ...this.state.csv,
+        csvFields,
+        entity,
+      },
+      isSelectDisabled: false,
+    }, () => {
+      if (!this.state.csv.id) {
+        this.props.createCsvConfig(this.state.csv);
+      } else {
+        this.props.saveCsvConfig(this.state.csv);
+      }
+    });
+  }
+
+  getOptions() {
+    return _.map(this.props.fieldConfig[this.state.csv.entity], v => ({
+      label: v.displayName,
+      value: v.id,
+    }));
+  }
+
+  getRelatedFieldValues(relatedField) {
+    return _.map(this.props.entity[relatedField], v => ({
+      label: v.name,
+      value: v.id,
+    }));
+  }
+
+  getFields(values, index) {
+    let hidden = true;
+    let hiddenRelated = true;
+    if (index) {
+      _.map(this.props.fieldConfig[this.state.csv.entity], (entity) => {
+        if (entity.id === values[index].fieldConfigId) {
+          if (entity.fieldType === 'DATE' || entity.fieldType === 'DATE_TIME') {
+            hidden = false;
+          }
+          if (entity.relatedEntity) {
```

Comment:
I think you should check the type here (there is type relation and relatedEntity will be specified for the collection too and there should be no mapping in this case)